AK/Slugify.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019-2020, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/DeprecatedString.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <cctype>
+
+namespace AK {
+
+inline ErrorOr<String> slugify(String const& _raw, char glue = '-')
+{
+    StringBuilder sb;
+    bool just_processed_space = false;
+
+    for (auto const& ch : _raw.bytes()) {
```

Comment:
Don't iterate a String's bytes. String holds text as utf-8, so a code-point can be more than 1 byte. Instead, iterate `_raw.code_points()` to get each code-point as a u32.

Reply:
I tried this. I am getting _empty_ output.

Reply:
Oh it was UAF bug. [here](https://discord.com/channels/830522505605283862/860165510263996427/1157364726116929598)